src/Messages/OWSMessageSender.m

```diff
@@ -42,10 +42,12 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
-#define SYSTEM_VERSION_GREATER_THAN_OR_EQUAL_TO(major, minor)                                                          \
-    ([[NSProcessInfo processInfo]                                                                                      \
-        isOperatingSystemAtLeastVersion:(NSOperatingSystemVersion){                                                    \
-                                            .majorVersion = major, .minorVersion = minor, .patchVersion = 0 }])
+void AssertIsOnSendingQueue()
+{
+    if (SYSTEM_VERSION_GREATER_THAN_OR_EQUAL_TO(10, 0)) {
+        dispatch_assert_queue([OWSDispatch sendingQueue]);
```

Comment:
I understand the risks of dispatch_sync. But I don't understand your concern with dispatch_assert_queue.

What is the danger introduced by adding an *assert*? The *assert* doesn't do any dispatch -
 - it just blows up when violated in DEBUG. Right?